### ISSUE TITLE
Input fields in dialogs need to be labeled

### ISSUE THREAD

**Issue by clapierre** [jupyterlab/jupyterlab#6581]
From web4all hackathon https://github.com/diagram-codesprint/jupyterlab/issues/12
## Description

Dialogs are not accessible. When opened, nothing is read beside the "Ok" button (which has the focus. They need to read the dialog title.

Additionally, those dialogs with input fields need labels on the input fields.

The fix to make the dialog state their title was two-fold:
1. The title was made an `h1` and given an `id`.
2. The dialog uses aria-labelledby to point to that `id`.

The fix for input field was similar:
1. The label was given an `id`
2. The input filed or combo box then pointed to that `id`via `aria-labelledby`.

Note: there is some code duplication and it would be good to do some refactorings so that these kinds of things are picked up automatically by any new dialog. It would also make some display more uniform.

**Comment by tonyfast** [comment]
this issues relates to **Text Issue Area # 4** in https://github.com/jupyterlab/jupyterlab/issues/9399

**Comment by t03857785** [comment]
I think this is now resolved by #14560 with a slightly altered implementation

**Comment by krassowski** [comment]
Thank you, closing as resolved!

If anyone from the future believes there is a follow-up needed, please open a new issue with details (and referencing this one).
